ds/reactive/variable_test.go

```diff
@@ -135,3 +135,121 @@ func TestOnUpdateOnce(t *testing.T) {
 		require.Equal(t, 0, calledNewValue)
 	}
 }
+
+// TestVariable_Init tests the Init method of the variable type
+func TestVariable_Init(t *testing.T) {
+	varValue := 10
+	v := newVariable[int]().Init(varValue)
+	require.Equal(t, varValue, v.Get())
+}
+
+// TestVariable_InheritFrom tests the InheritFrom method of the variable type
+func TestVariable_InheritFrom(t *testing.T) {
+	parentVar := newVariable[int]().Init(10)
+	childVar := newVariable[int]()
+
+	unsubscribe := childVar.InheritFrom(parentVar)
+	defer unsubscribe()
+
+	parentVar.Set(20)
+	require.Equal(t, 20, childVar.Get())
+}
+
+// TestVariable_DeriveValueFrom tests the DeriveValueFrom method of the variable type
+func TestVariable_DeriveValueFrom(t *testing.T) {
+	sourceVar := newDerivedVariable[int](func(d DerivedVariable[int]) func() { return func() {} }, 10)
+	derivedVar := newVariable[int]()
+
+	teardown := derivedVar.DeriveValueFrom(sourceVar)
+	defer teardown()
+
+	require.Equal(t, 10, derivedVar.Get())
+	sourceVar.Set(20)
+	require.Equal(t, 20, derivedVar.Get())
+}
+
+// TestVariable_WithValue tests the WithValue method of the variable type
+func TestVariable_WithValue(t *testing.T) {
+	rVar := newVariable[int]()
+	rVar.Set(10)
+
+	setupCalled := false
+	teardownCalled := false
+
+	teardown := rVar.WithValue(
+		func(value int) (teardown func()) {
+			setupCalled = true
+			return func() { teardownCalled = true }
+		},
+	)
+	defer teardown()
+
+	rVar.Set(20)
+	require.True(t, setupCalled)
+	require.True(t, teardownCalled)
+}
+
+// TestVariable_WithNonEmptyValue tests the WithNonEmptyValue method of the variable type
+func TestVariable_WithNonEmptyValue(t *testing.T) {
+	rVar := newVariable[int]()
+	rVar.Set(10)
+	setupCalled := false
+
+	teardown := rVar.WithNonEmptyValue(
+		func(value int) (teardown func()) {
+			setupCalled = true
+			return func() {}
+		},
+	)
+	defer teardown()
+
+	rVar.Set(20)
+	require.True(t, setupCalled)
+}
+
+// TestVariable_OnUpdateOnce tests the OnUpdateOnce method of the variable type
+func TestVariable_OnUpdateOnce(t *testing.T) {
+	rVar := newVariable[int]()
+	rVar.Set(10)
+	callbackCalled := false
+
+	unsubscribe := rVar.OnUpdateOnce(func(oldValue, newValue int) {
+		callbackCalled = true
+	})
+	defer unsubscribe()
+
+	rVar.Set(20)
+	require.True(t, callbackCalled)
+}
+
+// TestReadableVariable_OnUpdateOnce_WithCondition tests OnUpdateOnce with a condition
+func TestReadableVariable_OnUpdateOnce_WithCondition(t *testing.T) {
+	rv := newVariable[int]()
+	rv.Set(0)
+
+	var oldValue, newValue int
+	conditionNotMet := true
+
+	// Define a condition that is not met
+	condition := func(oldVal, newVal int) bool {
+		return newVal > 5 // Only trigger if the new value is greater than 5
+	}
+
+	// Register the callback with the condition
+	unsubscribe := rv.OnUpdateOnce(func(o, n int) {
+		oldValue = o
+		newValue = n
+		conditionNotMet = false
```

Comment:
Should be a counter, and we should set several values, to check if it is only called once or not.